app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TweetsController < ApplicationController
+
+  #### TWEETS INDEX ACTION ####
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      @user = User.find(session[:user_id])
+      erb :'tweets/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  #### CREATE ACTIONS ####
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'tweets/new'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do 
+    if params[:content] == ""
+      redirect '/tweets/new'
+    else
+      user = current_user
+      @tweet = Tweet.create(content: params[:content], user_id: user.id)
+
+      redirect "/tweets/#{@tweet.id}"
+    end
+  end
+
+  #### SHOW/READ ACTION ####
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = find_tweet(params)
+      erb :'/tweets/show'
+    else
+      redirect '/login'
+    end
+  end
+
+  #### EDIT ACTIONS ####
+
+  get '/tweets/:id/edit' do
+    @tweet = find_tweet(params)
+    if logged_in? && @tweet.user_id == current_user.id
+      erb :'/tweets/edit'
+    else
+      redirect '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content] == ""
+      redirect "/tweets/#{params[:id]}/edit"
+    else
+      @tweet = find_tweet(params)
+      @tweet.content = params[:content]
```

Comment:
lets use the update method in ActiveRecord here so we don't need to use the save method 
```tweet.update(content: params[:content])```